File: src/main/java/com/pweb/eventhorizon/model/dto/UserDto.java
package com.pweb.eventhorizon.model.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.Email;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class UserDto {

    @NotEmpty(message = "First name is mandatory")
    @NotBlank(message = "First name is mandatory")
    private String firstName;

    @NotEmpty(message = "Last name is mandatory")
    @NotBlank(message = "Last name is mandatory")
    private String lastName;

    @NotEmpty(message = "Email is mandatory")
    @NotBlank(message = "Email is mandatory")
    @Email
    private String email;

    private List<EventDto> events;

    private List<CommentDto> comments;
}

File: Frontend/src/app/models/user-dto.ts
/* tslint:disable */
/* eslint-disable */
import { CommentDto } from './comment-dto';
import { EventDto } from './event-dto';
export interface UserDto {
  comments?: Array<CommentDto>;
  email: string;
  events?: Array<EventDto>;
  firstName: string;
  id?: string;
  lastName: string;
}
